waspc/src/Wasp/Lib.hs

```diff
@@ -71,3 +73,9 @@ findDotEnvFile waspDir = do
   let dotEnvAbsPath = waspDir SP.</> [relfile|.env|]
   dotEnvExists <- doesFileExist (SP.toFilePath dotEnvAbsPath)
   return $ if dotEnvExists then Just dotEnvAbsPath else Nothing
+
+findMigrationsDir :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe (Path' Abs Dir'))
+findMigrationsDir waspDir = do
+  let migrationsAbsPath = waspDir SP.</> [reldir|migrations|]
```

Comment:
Regarding path of migrations in the wasp (source) project -> do we have a central place where it is defined? Because here you are defining it yourself, you are saying it is in waspDir, /migrations. But, what I know for certain is that CLI right now is doing copying of migrations from generator up to the src directory (yes, the opposite direction of usual, that is a bit special situation that we have, that is also interesting to observe / think about), which means it also somehow "knows" where to write them. This means this piece of knowledge is now in two places -> so we should look how to make it so it has just one source of truth.

Reply:
Good idea on defining a single source of truth for the migration path to be used in both places. I can try to find a good place to put it under `src` that we can also leverage in `cli`. 👍🏻

Reply:
I added a new commit that has a single source of truth for each of the migrations dir in the (a) wasp dir and (b) output dir: https://github.com/wasp-lang/wasp/pull/383/commits/134d19debd07f04d56c2c731486706a2b25e96d0 It feels nice now to only have two instances of `[reldir|migrations|]` in the codebase. :D

One thing you will notice is when `src` and `target` became `PathAbsDir`. This was so the expression had the same types for the `if` and `else`. I think it used to work in the old code because the way they defined the relative migration path in the first line.

However, once you make them relative to different directories in `SP` they became different types and I got the following error:
```
Migrate.hs
/Users/shayne/dev/wasp/waspc/cli/Wasp/Cli/Command/Db/Migrate.hs:93:16: error:
    • Couldn't match type ‘Wasp.Generator.DbGenerator.DbMigrationsDir’
                     with ‘()’
      Expected type: StrongPath.Internal.Path
                       StrongPath.Internal.System Abs StrongPath.Internal.Dir'
        Actual type: StrongPath.Internal.Path
                       StrongPath.Internal.System
                       Abs
                       (Dir Wasp.Generator.DbGenerator.DbMigrationsDir)
    • In the expression: dbMigrationsDirInGenProjectDirAbs
      In the expression:
        if copyDirection == CopyMigDirUp then
            dbMigrationsDirInWaspProjectDirAbs
        else
            dbMigrationsDirInGenProjectDirAbs
      In an equation for ‘target’:
          target
            = if copyDirection == CopyMigDirUp then
                  dbMigrationsDirInWaspProjectDirAbs
              else
                  dbMigrationsDirInGenProjectDirAbs
   |
93 |           else dbMigrationsDirInGenProjectDirAbs
   |    
```

Probably some way to clean it up beyond my current Haskell knowledge. I tried using a type param for `(Dir d)`and defining the type explicitly, but I could not use that in a `let`, nor make it work in other contexts. Happy to explore any ideas you may have to improve that area though since it does not feel DRY. Thanks!